pkg/slayers/path/scion/raw.go

```diff
@@ -319,40 +319,63 @@ func (s *Raw) GetCurrentInfoField( /*@ ghost ubuf []byte @*/ ) (res path.InfoFie
 }
 
 // SetInfoField updates the InfoField at a given index.
-// @ requires  0 <= idx
+// @ requires 0 <= idx
 // @ requires sl.AbsSlice_Bytes(ubuf, 0, len(ubuf))
 // @ requires acc(s.Mem(ubuf), R20)
-// @ ensures  acc(s.Mem(ubuf), R20)
-// @ ensures sl.AbsSlice_Bytes(ubuf, 0, len(ubuf))
-// @ ensures   r != nil ==> r.ErrorMem()
-// contracts for IO-spec
+// pres for IO:
 // @ requires dp.Valid() && validPktMetaHdr(ubuf) && s.EqAbsHeader(ubuf)
-// @ ensures r == nil && idx == int(old(s.GetCurrINF(ubuf))) ==>
-// @ 	dp.Valid() && validPktMetaHdr(ubuf) && s.EqAbsHeader(ubuf)
-// @ ensures r == nil && idx == int(old(s.GetCurrINF(ubuf))) ==>
-// @ 	s.absPkt(dp, ubuf) == AbsSetInfoField(old(s.absPkt(dp, ubuf)), info.ToIntermediateAbsInfoField())
+// @ ensures  acc(s.Mem(ubuf), R20)
+// @ ensures  sl.AbsSlice_Bytes(ubuf, 0, len(ubuf))
+// @ ensures  r != nil ==> r.ErrorMem()
+// posts for IO:
```

Comment:
Just as a reminder: If we rename `contract for IO-spec` to `pres for IO` and `posts for IO` we should do everywhere.